### ISSUE TITLE
react-navigation reset does not immediately unmount screens from stack

### ISSUE THREAD

**Issue by eballeste** [react-community/react-navigation#2996]
### Current Behavior
- Trying to accomplish auth signout by resetting the navigation stack after a successful logout:
```
  ...
  yield put(
    NavigationActions.reset({
      index: 0, 
      actions: [ 
        NavigationActions.navigate({ routeName: 'SignedOut' })
      ],
    })
  );
  // causes re-renders, screens still mounted
  yield put({type: RESET_STATE});
  ...
````
- I try to reset the redux state which throws exceptions because the screens connected to the redux state from the react-navigation stack have not yet been unmounted, causing them to re-render and throw undefined exceptions. 

- I also tried to reset once user reaches the SignedOut screen in it's `componentDidMount` cycle  but unfortunately the screens unmount at some point well after `componentDidMount` is triggered. 

```
class SignedOut extends Component {  
  componentDidMount() {
    // screens are still mounted, throws exceptions
    this.props.dispatch(resetState());
  }
  ...
```

### Expected Behavior
- I would expect the old screens to unmount by the time I reach the SignedOut screen in order for me to successfully reset the redux-state or for the reset action to provide some type of callback after the screens unmount. 

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | 1.0.0-beta.11
| react-native     | 0.50.1
| node             | 7.10.0
| npm       | 4.2.0

**Comment by kelset** [comment]
Hi thanks for reporting; a few questions:

1. is it somewhat related to this https://github.com/react-community/react-navigation/issues/2291 ?
1. have you tried updating to latest release (`beta.19`)? Moreover, `beta.20` will have some juicy updates related to "navigation" lifecycle, if you are brave try out `master` - it may be that it's fixed.
1. are you using `redux-saga`? Asking because of those `yields`.

**Comment by eballeste** [comment]
I think it's unrelated because the reset is working, it's just that it
unmounts the screens way further down the line after the new stack is
mounted, I'll try to update my react-navigation package and test it out
tonight, thanks for your response!

**Comment by eballeste** [comment]
Also, yes, I am using redux saga to trigger these events.

**Comment by eballeste** [comment]
After thinking about this for a long time, I figured out that maybe I should have been focusing on the errors thrown instead of why I was getting errors. (I did learn a lot though).

Although figuring out how to listen for when all the screens are completely unmounted after calling a reset would have been super helpful, it would have just been a shortcut to bypass the real issue, the initialState for certain branches of my redux state was wrong. After correcting this, no more errors, no matter when react-navigation decides to unmount the old screens.
